dtm0/doc/dtx.txt

```diff
@@ -0,0 +1,441 @@
+--------------------------------------------------------------------------------
+# Overview
+
+This document defines a DLD for the client-side
+state machines touched by DTM0. It has the following sections:
+
+* Component diagram - defines the scope.
+* With/Without DTM enabled - defines states and high-level view of 
+  the transitions.
+* Low-level details on events - a human-readable description of the
+  previous section.
+* DTX connections in psudo-code - describes what needs to be changed
+  in the existing code and how the changes are connected to DTM.
+* Q&A - a list of questions to clarify design decisions.
+
+--------------------------------------------------------------------------------
+# Component diagram
+
+      * Client-side SMs
+     \|/
++------------------+                       +-------------+
+| M0Client/DIX/CAS |                       | DTM Service |
++------------------+                       +-------------+
+        |                                       |
+        +---+-----------------------------------+ <-* "Interconnections" 
+            |
+           \|/
+        +----------+
+        | DTM0 log |
+        +----------+
+
+The scope of this document is limited by "Interconnections" and 
+"Client-side SMs".
+
+--------------------------------------------------------------------------------
+# Without DTM enabled
+
+This section is an overview of how Client-side SMs are defined
+right now.
+
+SM states:
+
+M0client-op: Init -- Launched -- Executed -- Stable
+
+DIX-req: Init --- ... (discovery) ... -- InProgress(N) -- Final
+
+CAS-req: Init -- Sent -- Final
+
+SM Events:
+
+launch(m0c):
+	m0c: Init -> Launched
+	dix: Init
+	cas: Init
+	ast: async call dix_send
+
+dix_send(dix):
+	dix: Init -> InProgress
+	cas: Init -> Sent
+	rpc: call cas_repl on reply
+
+cas_repl(cas[0]):
+	cas: Sent -> Final
+	dix: InProgress, nr_received++
+
+cas_repl(cas[N-1]):
+	cas: Sent -> Final
+	dix: InProgress -> Final
+	m0c: Launched -> Executed -> Stable
+
+--------------------------------------------------------------------------------
+# With DTM enabled
```

Comment:
General notice:

Please add FAILED states also.

Reply:
Added (partially) into some of the machines